gc/base/standard/ConcurrentGCIncrementalUpdate.hpp

```diff
@@ -1,83 +1,233 @@
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#if !defined(CONCURRENTGCINCREMENTALUPDATE_HPP_)
+#define CONCURRENTGCINCREMENTALUPDATE_HPP_
+
+#include "omrcfg.h"
+#include "modronopt.h"
+
+#include "omr.h"
+#include "OMR_VM.hpp"
+#if defined(OMR_GC_MODRON_CONCURRENT_MARK)
+
+#include "ConcurrentGC.hpp"
+#include "ConcurrentCardTable.hpp"
+
+/**
+ * @name Concurrent mark card cleaning factor
+ * @{
+ */
+#define INITIAL_CARD_CLEANING_FACTOR_PASS1_1 ((float)0.5)
+#define INITIAL_CARD_CLEANING_FACTOR_PASS1_8 ((float)0.05)
+#define INITIAL_CARD_CLEANING_FACTOR_PASS1_10 ((float)0.05)
+
+#define INITIAL_CARD_CLEANING_FACTOR_PASS2_1 ((float)0.1)
+#define INITIAL_CARD_CLEANING_FACTOR_PASS2_8 ((float)0.01)
+#define INITIAL_CARD_CLEANING_FACTOR_PASS2_10 ((float)0.01)
+
+#define MAX_CARD_CLEANING_FACTOR_PASS1_1 ((float)0.8)
+#define MAX_CARD_CLEANING_FACTOR_PASS1_8 ((float)0.2)
+#define MAX_CARD_CLEANING_FACTOR_PASS1_10 ((float)0.2)
+
+#define MAX_CARD_CLEANING_FACTOR_PASS2_1 ((float)0.5)
+#define MAX_CARD_CLEANING_FACTOR_PASS2_8 ((float)0.1)
+#define MAX_CARD_CLEANING_FACTOR_PASS2_10 ((float)0.1)
+
+/**
+ * @}
+ */
+
+/**
+ * @name Concurrent mark card cleaning threshold
+ * @{
+ */
+#define CARD_CLEANING_THRESHOLD_FACTOR_1 ((float)4.0)
+#define CARD_CLEANING_THRESHOLD_FACTOR_8 ((float)3.0)
+#define CARD_CLEANING_THRESHOLD_FACTOR_10 ((float)1.5)
+
+/**
+ * @}
+ */
+
+/**
+ * @name Concurrent mark misc definitions
+ * @{
+ */
+#define CARD_CLEANING_HISTORY_WEIGHT ((float)0.7)
+#define BYTES_TRACED_IN_PASS_1_HISTORY_WEIGHT ((float)0.8)
+#define ALL_BYTES_TRACED_IN_PASS_1 ((float)1.0)
+/**
+ * @}
+ */
+
+/**
+ * @todo Provide class documentation
+ * @ingroup GC_Modron_Standard
+ */
+class MM_ConcurrentGCIncrementalUpdate : public MM_ConcurrentGC
+{
+	/*
+	 * Data members
+	 */
+private:
+	MM_ConcurrentCardTable *_cardTable;	/**< pointer to Cards Table */
+
+	/* Concurrent card cleaning statistics */
+	float _cardCleaningFactorPass1;
+	float _cardCleaningFactorPass2;
+	float _maxCardCleaningFactorPass1;
+	float _maxCardCleaningFactorPass2;
+	float _cardCleaningThresholdFactor;
+	float _bytesTracedInPass1Factor;
+
+	uintptr_t _bytesToCleanPass1;
+	uintptr_t _bytesToCleanPass2;
+	uintptr_t _bytesToTracePass1;
+	uintptr_t _bytesToTracePass2;
+	uintptr_t _allocToTraceRateCardCleanPass2Boost;
+	uintptr_t _totalTracedAtPass2KO;
+	uintptr_t _totalCleanedAtPass2KO;
+	uintptr_t _traceTargetPass1;
+	uintptr_t _traceTargetPass2;
+
+	bool _pass2Started;
+	bool  _secondCardCleanPass;
```

Comment:
ideally all new variables should be annotated

Reply:
I've added to the existing comment at the start, where these variables are declared

Reply:
just for the record - individual description of class variables in format `/**< ... */` is formally required for support documenting utility (yes, we never run probably)